client/navigation/utils.ts

```diff
@@ -43,50 +82,69 @@ export const getMatchScore = ( location, itemUrl, itemExpression = null ) => {
 	return ( decodeURIComponent( href ).match( regexp ) || [] ).length;
 };
 
-/**
- * Get a default expression to match the path and provided params.
- *
- * @param {string} url URL to match.
- * @return {string} Regex expression.
- */
-export const getDefaultMatchExpression = ( url ) => {
-	const escapedUrl = url.replace( /[-\/\\^$*+?.()|[\]{}]/gi, '\\$&' );
-	const [ path, args, hash ] = escapedUrl.split( /\\\?|#/ );
-	const hashExpression = hash ? `(.*#${ hash }$)` : '';
-	const argsExpression = args
-		? args.split( '&' ).reduce( ( acc, param ) => {
-				return `${ acc }(?=.*[?|&]${ param }(&|$|#))`;
-		  }, '' )
-		: '';
-	return '^' + path + argsExpression + hashExpression;
-};
+interface wcNavigation {
+	menuItems: Item[];
+	rootBackLabel: string;
+	rootBackUrl: string;
+	historyPatched: boolean;
+}
+
+declare global {
+	interface Window {
+		wcNavigation: wcNavigation;
+	}
+}
+
+interface wcNavigation {
+	menuItems: Array< Item >;
+	rootBackLabel: string;
+	rootBackUrl: string;
+	historyPatched: boolean;
+}
+
+declare global {
+	interface Window {
+		wcNavigation: wcNavigation;
+	}
+}
 
 /**
  * Adds a listener that runs on history change.
  *
  * @param {Function} listener Listener to add on history change.
  * @return {Function} Function to remove listeners.
  */
-export const addHistoryListener = ( listener ) => {
+export const addHistoryListener = ( listener: () => void ): ( () => void ) => {
```

Comment:
@joelclimbsthings I had to mess around with this function and tried to find your PR that introduced this code so I could check against the test instructions. I can't find it because this code was moved over from the old Navigation repo. Do you remember what this code is supposed to do? I think it was so that persisted queries remained intact when navigating around, including Home screen. Does that ring a bell?

Reply:
Indeed, very large ones! Although that function, `addhistoryListener()` preexisted that, here's the PR for the persisted queries: https://github.com/woocommerce/woocommerce-admin/pull/5883

I also used it for the nav folding behavior as well: https://github.com/woocommerce/woocommerce-admin/pull/5578.

Reply:
Thanks @joelclimbsthings

https://github.com/woocommerce/woocommerce-admin/pull/5578 is working, and https://github.com/woocommerce/woocommerce-admin/pull/5883 is testing the same as `main`. Queries persist from report to report, but if you navigate to Homescreen, they don't persist. This is with the new Nav. Using it without Nav going to Homescreen keeps the queries when you go back to a report.

Is this a known thing? I can't remember if we accepted this as fact or if it ever worked.

Reply:
Ah, yes--that should work as you expected. Looks like something 💔 . I'll create a new issue and try to add a test, but shouldn't hold up this PR.

Reply:
Thanks @joelclimbsthings !